flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/EdgeManager.java

```diff
@@ -50,14 +50,11 @@ public void connectPartitionWithConsumerVertexGroup(
 
         checkNotNull(consumerVertexGroup);
 
-        final List<ConsumerVertexGroup> consumers =
-                getConsumerVertexGroupsForPartitionInternal(resultPartitionId);
-
-        // sanity check
         checkState(
-                consumers.isEmpty(), "Currently there has to be exactly one consumer in real jobs");
+                getConsumerVertexGroupForPartition(resultPartitionId) == null,
+                "Currently one partition can have at most one consumer group");
 
-        consumers.add(consumerVertexGroup);
+        partitionConsumers.put(resultPartitionId, consumerVertexGroup);
```

Comment:
I'm wondering whether `checkState(partitionConsumers.putIfAbsent(resultPartitionId) == null, "Currently one partition can have at most one consumer group");` is better here?

Reply:
Yes, it is better. I will apply this change .